### ISSUE TITLE
`1 == 1` produces no warning while `1 is 1` produces CS8520

### ISSUE THREAD

**Issue by rampaa** [dotnet/roslyn#82869]
**Version Used**: 
Visual Studio 2026 v18.4.1
.NET 9

**Steps to Reproduce**:

1. See the following minimal code sample:
```csharp
    public static void Test()
    {
        if (1 == 1) // Does not produce a warning like CS8520
        {
            Console.WriteLine("Test 1");
        }

        if (1 is 1) // CS8520
        {
            Console.WriteLine("Test 1");
        }
    }
```

**Expected Behavior**:
`1 == 1` should raise a warning like how `1 is 1` raises CS8520

**Actual Behavior**:
No warning is raised for `1 == 1`

**Comment by jjonescz** [comment]
That warning is a feature of patterns (which are in use with `is` but not with `==`), so that's likely expected. We could implement a subset of such warnings that would apply to `==` I guess (it would probably need a language design proposal in https://github.com/dotnet/csharplang), although that could be also just a CA analyzer in https://github.com/dotnet/sdk.
